_posts/2019-08-18-view-encapsulation-w-angularze.md

```diff
@@ -0,0 +1,426 @@
+---
+layout:    post
+title:     "View Encapsulation w Angularze - czyli o kapsułkowaniu słów kilka"
+published: true
+date:     2019-08-18 08:00:00 +0100
+author:    mhoja
+tags:
+    - angular
+    - view encapsulation
+    - kapsułkowanie
+---
+
+Tworząc komponenty w Angularze mamy możliwość zarządzania kapsułkowaniem styli - czyli tym jak style z jednego komponentu wpływają na inne komponenty.  
+Zanim omówimy kapsułkowanie, wtyjaśnijmy w kilku słowach czym jest **Shadow DOM**.
+
+## Shadow DOM
+
+**Shadow DOM** wprowadza enksapsulację do DOM'u. Pozwala to odseparować styl i kod potrzebny do wyświetlenia od dokumentu, w którym znajduje się dany element. Przykładem może być np. znacznik HTML `<video>`
+
+```html
+<video width="320" height="240">
+    <source src="movie.mp4" type="video/mp4">
+    <source src="movie.ogg" type="video/ogg">
+</video>
+```
+
+Po włączeniu opcji wyświetlania Shadow Root w przeglądarce:  
+![Devtools Configuration](/assets/img/posts/2019-xx-xx-view-encapsulation/devtools_config_shadow_dom.jpg)
+
+możemy zobaczyć z czego tak naprawdę składa się znacznik `<video>`:  
+
+```html
+<video width="320" height="240">
+    #shadow-root
```

Comment:
to jest zapis stosowany przez chrome, warto to zaznaczyć

Reply:
Dopisałem przy screenie z DevTools że to przykład z Chrome - według mnie tyle wystarczy.